src/main/scala/ai/eto/rikai/sql/model/mlflow/MlflowCatalog.scala

```diff
@@ -0,0 +1,142 @@
+  override def createModel(model: Model): Model = {
+    val name = mlflowClient.createModel(
+      model.name,
+      Map() ++ model.flavor.map(ModelFlavorKey -> _)
+    )
+    new SparkUDFModel(
+      name,
+      s"mlflow://$name",
+      "<anonymous>",
+      model.flavor
+    )
+  }
+
+  /** Return a list of models available for all Sessions */
+  override def listModels(): Seq[Model] = {
+    val response = mlflowClient.searchRegisteredModels()
+    response.getRegisteredModelsList.asScala
+      .map(model => {
+        model.getLatestVersionsCount match {
+          case 0 =>
+            None
+          case _ =>
+            val latestVersion = model.getLatestVersions(0)
+            val tagsMap = latestVersion.getTagsList.asScala
+              .map(t => t.getKey -> t.getValue)
+              .toMap
+            val name = model.getName
+            if (
+              true
+//              TODO enable this after https://github.com/eto-ai/rikai/pull/351 finished
+//              && tagsMap.contains(ArtifactPathKey)
+            ) {
+              val flavor = tagsMap.getOrElse(ModelFlavorKey, "")
+              Some(
+                new SparkUDFModel(
+                  name,
+                  s"mlflow://$name",
+                  "<anonymous>",
+                  flavor
+                )
+              )
+            } else {
+              None
+            }
+        }
+      })
+      .filter(m => m.isDefined)
+      .map(m => m.get)
+  }
+
+  /** Check a model with the specified name exists.
+    *
+    * @param name is the name of the model.
+    */
+  override def modelExists(name: String): Boolean = {
+    mlflowClient.getModel(name).nonEmpty
+  }
+
+  /** Get the model with a specific name.
+    *
+    * @param name is a qualified name pointed to a Model.
+    * @return the model
+    */
+  override def getModel(name: String): Option[Model] = {
+    mlflowClient.getModel(name).map { modelVersion =>
+      val tagsMap = modelVersion.getTagsList.asScala
+        .map(t => t.getKey -> t.getValue)
+        .toMap
+      new SparkUDFModel(
+        name,
+        s"mlflow://$name",
+        "<anonymous>",
+        tagsMap.get(ModelFlavorKey)
+      )
+    }
+  }
+
+  /** Drops a model with a specific name
+    *
+    * @param name the model name
+    * @return true of the model is dropped successfully. False otherwise.
+    */
+  override def dropModel(name: String): Boolean = {
+    try {
+      mlflowClient.deleteModel(name)
+      true
+    } catch {
+      case e: Exception =>
+        e.printStackTrace()
+        false
+    }
+  }
+}
+
+object MlflowCatalog {
+
+  val TrackingUriKey = "rikai.sql.ml.registry.mlflow.tracking_uri"
+
+  val ArtifactPathKey = "rikai.model.artifact_path"
+  val ModelFlavorKey = "rikai.model.flavor"
+  val OutputSchemaKey = "rikai.output.schema"
+  val SpecVersionKey = "rikai.spec.version"
+  val PreProcessingKey = "rikai.transforms.pre"
+  val PostProcessingKey = "rikai.transforms.post"
```

Comment:
for string literal, how about POST_PROCESSING_KEY

Reply:
updated